src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -154,4 +166,104 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
+            distributedApplicationLogger.LogError("Failed to get the dashboard logs.");
+            return;
+        }
+
+        if (dashboardResourceId is null)
+        {
+            return;
+        }
+
+        var defaultDashboardLogger = loggerFactory.CreateLogger("Aspire.Hosting.Dashboard");
+
+        var loggerCache = new Dictionary<string, ILogger>();
+
+        try
+        {
+            // We turned on the JSON formatter for the logger, so we can log the log lines as JSON.
+            await foreach (var batch in resourceLoggerService.WatchAsync(dashboardResourceId).ConfigureAwait(false))
+            {
+                foreach (var logLine in batch)
+                {
+                    try
+                    {
+                        var logMessage = JsonSerializer.Deserialize(logLine.Content, DashboardLogMessageContext.Default.DashboardLogMessage);
+
+                        if (logMessage is not null)
+                        {
+                            var logLevel = logMessage.LogLevel switch
+                            {
+                                "Information" => LogLevel.Information,
+                                "Debug" => LogLevel.Debug,
+                                "Warning" => LogLevel.Warning,
+                                "Error" => LogLevel.Error,
+                                "Critical" => LogLevel.Critical,
+                                "Trace" => LogLevel.Trace,
+                                _ => LogLevel.Information
+                            };
+
+                            if (!loggerCache.TryGetValue(logMessage.Category, out var logger))
+                            {
+                                var categoryTrimmed = logMessage.Category.StartsWith("Aspire.Dashboard.") ?
+                                    logMessage.Category["Aspire.Dashboard.".Length..] : logMessage.Category;
+
+                                loggerCache[logMessage.Category] = logger = loggerFactory.CreateLogger($"Aspire.Hosting.Dashboard.{categoryTrimmed}");
+                            }
+
+                            logger.Log(logLevel, logMessage.EventId, logMessage.Message, null, (s, _) => s);
+                        }
+                    }
+                    catch (JsonException)
+                    {
+                        // Failed to parse, it's not JSON, just log the content as is.
+                        var level = logLine.IsErrorMessage ? LogLevel.Error : LogLevel.Information;
+                        defaultDashboardLogger.Log(level, 0, logLine.Content, null, (s, _) => s);
+                    }
+                }
+            }
+        }
+        catch (Exception ex)
+        {
+            defaultDashboardLogger.LogError(ex, "Error logging dashboard logs.");
+        }
+    }
+}
+
+internal class DashboardLogMessage
+{
+    public string Timestamp { get; set; } = string.Empty;
+    public int EventId { get; set; }
+    public string LogLevel { get; set; } = string.Empty;
```

Comment:
```suggestion
    public LogLevel LogLevel { get; set; }
```

Why not use the enum?

Reply:
Doesn't work as we get a string from the external process that needs to be converted to an enum.

Reply:
Why can't you use `JsonStringEnumConverter<LogLevel>`?

Reply:
overhead? I can write an switch statement and avoid all of the overhead.

Reply:
> I can write a switch statement and avoid all of the overhead.

Following that same logic... you might as well use a Utf8JsonReader.

Reply:
Look at all of that crazy code in the converter compared to this switch 😄

Reply:
To this code, it is just 1 attribute on a property vs a 10 line switch.

As for runtime overhead, I wonder what performs better - doing an alloc-free enum parsing (what the converter does) or allocating a string just to switch over it...?

Reply:
Done